### ISSUE TITLE
UI interface improvements

### ISSUE THREAD

**Issue by robx** [robx/pzprjs#99]
Similar to #1 (which deals with the individual puzzle applets), this is to collect ideas for how the general UI (menus, toolbox, background etc.) could be improved.

**Comment by robx** [comment]
- solving tall puzzles is painful, particularly once the forward/back buttons and the top of the puzzle aren't visible at once; not sure how to solve this
- the border and background gets in the way / feels out of place once the window gets very narrow, e.g. on mobile phones; possible strip the background below some width
- the custom elements in the toolbox are messy (e.g. plain text "Input Mode | Auto ...")
- tool box takes too much space on mobile, would probably best be hidden by default with an easier toggle than currently
- there's redundant entries in the menu, e.g. clearing answer; remove those
- URL export should have a copy-to-clipboard button

**Comment by robx** [comment]
leaving trial mode should highlight the first trial change in some way

**Comment by robx** [comment]
It would make sense to have dedicated touch-device and "desktop" input modes. E.g., auxiliary dots in star battle have moved to right-click because they don't work well with touch, but with mouse they make more sense on left click. Fillomino is hard to play on mobile because you can draw borders and aux marks both by dragging, but that's fine with a mouse.

**Comment by robx** [comment]
the timer could be improved; pzprjs made a change that i don't understand, but some ideas:
- don't start timer until puzzle is shown (when opening in new background tab); or maybe condense long periods of inactivity
- offer a manual timer with start/stop buttons
- disable autocheck and thus timer stop for variant puzzles (by passing a variant url parameter)

**Comment by robx** [comment]
a way to indicate that a puzzle is a variant; I'm hesitant to allow inputting arbitrary text or links though, since that would hurt language-neutrality

perhaps a plain `variant` query parameter that would toggle a variant mode, with an optional `variant=<variant-ref>` link to some external definition of the type.

**Comment by robx** [comment]
improve handling of built-in variants (e.g., aquarium rule variant)

- the variant rule shouldn't be displayed as a disabled option in play mode; instead there should just be an explanatory sentence
- would be good to have a separate section in the toolbox for these

**Comment by x-sheep** [comment]
> * tool box takes too much space on mobile, would probably best be hidden by default with an easier toggle than currently

The toolbox is actually much more important on touch screens than when playing with a mouse, since you can't right click, and input is much less accurate (e.g. I frequently draw border lines instead of aux. lines or vice versa). I find myself constantly changing input modes when playing on my phone, while never needing it when using a desktop.

**Comment by robx** [comment]
> > * tool box takes too much space on mobile, would probably best be hidden by default with an easier toggle than currently
> 
> The toolbox is actually much more important on touch screens than when playing with a mouse, since you can't right click, and input is much less accurate (e.g. I frequently draw border lines instead of aux. lines or vice versa). I find myself constantly changing input modes when playing on my phone, while never needing it when using a desktop.

Right, input mode switching should stay there. Probably the right thing to do would be to move more of the options from the toolbox to the menu, and cut the toolbox down to input mode only.

**Comment by ropeko** [comment]
I'm rather new to the puzz.link. What I miss a bit is the possibility to go back to the list of puzzles from https://puzz.link/p to https://puzz.link/list.html. Of course it is possible to click back. But it is not that intuitive after working for some time on a puzzle.

**Comment by robx** [comment]
> I'm rather new to the puzz.link. What I miss a bit is the possibility to go back to the list of puzzles from https://puzz.link/p to https://puzz.link/list.html. Of course it is possible to click back. But it is not that intuitive after working for some time on a puzzle.

Maybe it would be best to get rid of the list page entirely, and instead open new editor windows with a list that you can choose from? Then there could be an entry in the menu that says "Create new puzzle" or similar to open a new window which starts out with the list of types.

**Comment by ropeko** [comment]
> > I'm rather new to the puzz.link. What I miss a bit is the possibility to go back to the list of puzzles from https://puzz.link/p to https://puzz.link/list.html. Of course it is possible to click back. But it is not that intuitive after working for some time on a puzzle.
> 
> Maybe it would be best to get rid of the list page entirely, and instead open new editor windows with a list that you can choose from? Then there could be an entry in the menu that says "Create new puzzle" or similar to open a new window which starts out with the list of types.

The list page is ok in my opinion. Maybe a search field would be nice. We could also just add target _blank to the links so that it will be opened in a new tab.
Furthermore I don’t understand the names „Nigun“, „Omopa“ etc. What exactly do they mean?
